### PR TITLE
feat(TBD-13647): allow switch from transient to interactive cluster

### PR DESCRIPTION
**Pull request by AlixMetivier-zz**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/AbstractSparkDistribution.java [hunk 1]
```diff
@@ -146,6 +147,10 @@ public boolean doSupportHBase1x() {
     public List<EDatabriksCloudProvider> getSupportCloudProviders() {
         return Arrays.asList(EDatabriksCloudProvider.values());
     }
+    
+    public List<EDatabriksClusterType> getClusterTypes() {
```
Comment (lbourgeois) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/AbstractSparkDistribution.java:151]: At some point we will need to review why we have some Databricks specific stuff in the AbstractSparkDistribution

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/EDatabriksClusterType.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+//
+// You should have received a copy of the agreement
+// along with this program; if not, write to Talend SA
+// 9 rue Pages 92150 Suresnes, France
+//
+// ============================================================================
+package org.talend.hadoop.distribution.constants.databricks;
+
+import org.talend.hadoop.distribution.i18n.Messages;
+public enum EDatabriksClusterType {
```
Comment (lbourgeois) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/EDatabriksClusterType.java:16]:
```suggestion
public enum EDatabricksClusterType {

```
There is same typo on other already existing enums. Not mandatory to change them in this feature

File: main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java [hunk 3]
```diff
@@ -826,6 +914,19 @@ private List<String> getProviders() {
         }
         return providerLableNames;
     }
+    
+    private List<String> getClusterTypes() {
+        List<String> providerLableNames = new ArrayList<String>();
```
Comment (lbourgeois) [inline main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java:919]:
```suggestion
        List<String> providerLabelNames = new ArrayList<String>();
```

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/EDatabriksClusterType.java (hunk 2 above)
Comment (AlixMetivier) [reply main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/EDatabriksClusterType.java:16]: ah good catch i'll correct them so its done

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/AbstractSparkDistribution.java (hunk 1 above)
Comment (AlixMetivier) [reply main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/AbstractSparkDistribution.java:151]: maybe it should be renamed to AbstractUniversalDistribution then :D

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/EDatabricksClusterType.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+// along with this program; if not, write to Talend SA
+// 9 rue Pages 92150 Suresnes, France
+//
+// ============================================================================
+package org.talend.hadoop.distribution.constants.databricks;
+
+import org.talend.hadoop.distribution.i18n.Messages;
+public enum EDatabricksClusterType  {
+
+    TRANSIENT(Messages.getString("EDatabricksClusterType .TRANSIENT"), "\"TRANSIENT\""),
```
Comment (lbourgeois) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/EDatabricksClusterType.java:18]:
```suggestion
    TRANSIENT(Messages.getString("EDatabricksClusterType.TRANSIENT"), "\"TRANSIENT\""),

```

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/EDatabricksClusterType.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    INTERACTIVE(Messages.getString("EDatabricksClusterType .INTERACTIVE"), "\"INTERACTIVE\"");
```
Comment (lbourgeois) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/constants/databricks/EDatabricksClusterType.java:19]:
```suggestion
    INTERACTIVE(Messages.getString("EDatabricksClusterType.INTERACTIVE"), "\"INTERACTIVE\"");

```

File: main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java (hunk 3 above)
Comment (lbourgeois) [reply main/plugins/org.talend.repository.hadoopcluster/src/org/talend/repository/hadoopcluster/ui/StandardHCInfoForm.java:919]: Up for the typo on Label

**Review by lbourgeois — DISMISSED** [review]
LGTM

**Review by lbourgeois — APPROVED** [review]
LGTM

**Pull request merged**